### ISSUE TITLE
Trackmania Forever detects wrong amount of VRAM

### ISSUE THREAD

**Issue by lorenz** [iXit/Mesa-3D#323]
Trackmania Forever detects only 128MiB of VRAM (see Configuration -> Advanced on the top right). The GPU physically has 4GiB. This causes Trackmania to use very low-quality textures. This is reproducible using a 128MiB VRAM override on WineD3D.

**Comment by mirh** [comment]
[Could be actually](https://pcgamingwiki.com/wiki/TrackMania_United_Forever#Low-res_textures_with_2_GB.2B_VRAM_GPUs) a problem with the game itself..

**Comment by lorenz** [comment]
I don't think so since it detects only 128MiB

**Comment by axeldavy** [comment]
How much memory do you force advertise wiith wine ?

**Comment by lorenz** [comment]
1024MiB/1GiB

**Comment by axeldavy** [comment]
if you say 3GB for example, does it still work ?

**Comment by lorenz** [comment]
Yes, on WineD3D it detects 3GiB just fine. 4GiB on the other hand results in a detection of 0 since it overflows the Game's 32 bit unsigned integer.
